### DISCUSSION TITLE
Customers can see order history for which they are not explicitly notified.

### DISCUSSION THREAD

**Discussion by mavelo-llc** [opencart/opencart#14184] [General]
Is this intentional? If you don't want to notify customers about status updates (eg: internal purposes, not customer-friendly, etc), you wouldn't want them to see it in their account area order history either. Thoughts?

**Comment by mavelo-llc** [comment]
Note: We'll be disabling unnotified history display in the account area using an event trigger, but wanted to get some community input on that scenario,

**Comment by hayden-t** [comment]
I think the thing is it was never meant to be private, just what the switch says, either to be notified or not, a private note could be a third option of use.

**Comment by ADDCreative** [comment]
The comments were private if notify wasn't checked in all versions before 4.0.2.2. I think it's a bug introduced by https://github.com/opencart/opencart/commit/c6bf54bbd8dbb5c4fdacf8b7bc47da66ded49567#diff-e5e9507491753eded8db7120106f2e12ad031d6682bf2c2fbaf75d1a86660db9 when the get histories was moved to a separate method.

The test for ```$result['notify']``` was missed of the comment.
https://github.com/opencart/opencart/blob/564e9f1848b4184d0828bd3f3c1a09e226db3c9b/upload/catalog/controller/account/order.php#L438
Should probably of been.
```'comment'    => $result['notify'] ? nl2br($result['comment']) : '',```

**Reply by hayden-t** [reply]
oh well that would be something different. and i think maybe the terminology used could have led to the mixup

**Reply by mavelo-llc** [reply]
I thought they were before but didn't have an older install to check right away.

**Comment by mavelo-llc** [comment]
This issue is present in the event/notification system as well. I'll have to do some digging to find all related areas and submit a pull when I have some free time.
